quit/basicfunction.py

```diff
@@ -0,0 +1,147 @@
+import typing
+
+import numpy as np
+from scipy import linalg
+
+Array1D = typing.Union[np.ndarray, typing.Sequence[complex]]
+
+
+def ket(psi: Array1D) -> np.ndarray:
+    """Create a ket |psi> as a vertical vector.
+
+    :param psi: Sequence of complex numbers.
+    :return: A vertical vector as numpy array.
+    """
+    psi = np.asarray(psi)
+
+    if len(psi.squeeze().shape) != 1:
+        raise ValueError("Wrong parameter psi.")
+    return psi.reshape((-1, 1))
+
+
+def basis(dim: int, index: int) -> np.ndarray:
+    """Create computational basis of dimension dim.
+
+    :param dim: Dimension of vector space.
+    :param index:
+    :return: A basis vector |d, i>.
+    """
+    if index >= dim:
+        raise ValueError("Parameter index out of range.")
+
+    vec = np.zeros(dim)
+    vec[index] = 1
+    return vec
+
+
+def dagger(matrix: np.ndarray) -> np.ndarray:
+    """Create conjugate transpose of a given matrix.
+
+    :param matrix: A matrix given by numpy array.
+    :return: An operator dagger as concatanation conjugate and transpose.
+    """
+    return np.transpose(np.conjugate(matrix))
+
+
+def bra(psi: Array1D) -> np.ndarray:
+    """Create bra <psi| as a horizontal vector.
+
+    :param psi: Sequence of complex numbers.
+    :return: A horizontal vector as numpy array satisfying <psi| = |psi>^+
+    """
+    return dagger(ket(psi))
+
+
+def braket(psi: Array1D, phi: Array1D) -> complex:
+    """Create braket as an inner product between vectors psi and phi.
+
+    :param psi: Sequence of complex numbers.
+    :param phi: Sequence of complex numbers.
+    :return: Inner product between vectors psi and phi.
+    """
+    return complex(bra(psi) @ ket(phi))
+
+
+def ketbra(psi: Array1D, phi: Array1D) -> np.ndarray:
+    """Compute outer product |psi><phi|.
+
+    :param psi: Sequence of complex numbers.
+    :param phi: Sequence of complex numbers.
+    :return: A matrix |psi><phi|
+    """
+    return ket(psi) @ bra(phi)
+
+
+def normaliced_vector(psi: Array1D) -> np.ndarray:
+    """Create a normalized vector |psi>.
+
+    :param psi: Sequence of complex numbers.
+    :return: A normaliced vector |psi> / |||psi>||.
+    """
+    return ket(psi / linalg.norm(psi))
+
+
+def proj(psi: Array1D) -> np.ndarray:
+    """Create a projector ontto normalized vector |psi>.
+
+    :param psi: Sequence of complex numbers.
+    :return: A matrix |psi><psi|.
+    """
+    return ket(normaliced_vector(psi)) @ bra(normaliced_vector(psi))
+
+
+def res(matrix: np.ndarray) -> np.ndarray:
+    """Reshaping maps matrix into a vector row by row. res(A) is equivalent to vec(A.T).
+
+    :param matrix: A matrix given by numpy array.
```

Comment:
Information that matrix is given by numpy array is redundant, as it is present in the docstring. A better description would be "A matrix to be reshaped".

The same goes for other functions with similar parameter description.